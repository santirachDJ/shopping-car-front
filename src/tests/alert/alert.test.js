import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import AlertShowError from "../../components/alert/alert.component";
import Alert from "emerald-ui/lib/Alert";
import { findElement } from "../utils/utilsEnzyme";

const waitForComponentToPaint = async (wrapper) => {
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 0));
    wrapper.update();
  });
};

describe("alert test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AlertShowError message="Product add" color="info" />);
    waitForComponentToPaint(wrapper);
  });

  afterEach(() => {
    wrapper = {};
  });

  it("Matches the snapshot", () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('Find element componenet', () => {
    expect(findElement(wrapper, Alert).exists()).toBe(true)
})
});
