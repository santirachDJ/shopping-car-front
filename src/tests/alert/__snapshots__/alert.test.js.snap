// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert test Matches the snapshot 1`] = `
Array [
  <AlertShowError
    color="info"
    message="Product add"
  />,
]
`;
